transfer/transfer-07-provider-push-http/README.md

```diff
@@ -0,0 +1,469 @@
+send counter offers in addition to just confirming or declining an offer.
+
+```bash
+curl -d '{
+           "connectorId": "http-push-provider",
+           "connectorAddress": "http://localhost:19194/api/v1/ids/data",
+           "protocol": "ids-multipart",
+           "offer": {
+             "offerId": "1:50f75a7a-5f81-4764-b2f9-ac258c3628e2",
+             "assetId": "assetId",
+             "policy": {
+               "uid": "231802-bb34-11ec-8422-0242ac120002",
+               "permissions": [
+                 {
+                   "target": "assetId",
+                   "action": {
+                     "type": "USE"
+                   },
+                   "edctype": "dataspaceconnector:permission"
+                 }
+               ],
+               "@type": {
+                 "@policytype": "set"
+               }
+             }
+           }
+         }' -X POST -H 'content-type: application/json' http://localhost:29193/api/v1/data/contractnegotiations \
+         -s | jq
+```
+
+Sample output:
+
+```json
+{
+  "createdAt": 1674585892398,
+  "id": "8ce50f33-25f3-42df-99e7-d6d72d83032c"
+}
+```
+
+### 8. Getting the contract agreement id
+
+After calling the endpoint for initiating a contract negotiation, we get a UUID as the response.
+This UUID is the ID of the ongoing contract negotiation between consumer and provider. The
+negotiation sequence between provider and consumer is executed asynchronously in the background by a
+state machine. Once both provider and consumer either reach the `confirmed` or the  `declined`
+state, the negotiation is finished. We can now use the UUID to check the current status of the
+negotiation using an endpoint on the consumer side.
+
+```bash
+curl -X GET "http://localhost:29193/api/v1/data/contractnegotiations/<contract negotiation id, returned by the negotiation call>" \
+    --header 'Content-Type: application/json' \
+    -s | jq
+```
+
+Sample output:
+
+```json
+{
+  "createdAt": 1674585892398,
+  "updatedAt": 1674585897476,
+  "contractAgreementId": "1:307a028a-b2b3-495e-ab6c-f6dad24dd098",
+  "counterPartyAddress": "http://localhost:19194/api/v1/ids/data",
+  "errorDetail": null,
+  "id": "8ce50f33-25f3-42df-99e7-d6d72d83032c",
+  "protocol": "ids-multipart",
+  "state": "CONFIRMED",
+  "type": "CONSUMER"
+}
+```
+
+### 9. Start the transfer
+
+Now that we have a contract agreement, we can finally request the file. In the request body, we need
+to specify which asset we want transferred, the ID of the contract agreement, the address of the
+provider connector and where we want the file transferred. You will find the request body below.
+Before executing the request, insert the contract agreement ID from the previous step. Then run :
+
+> keep in mind that, to make a transfer with a provider push method, the dataDestination type should
+> be any value different from the "HttpProxy".
+
+```bash
+curl -X POST "http://localhost:29193/api/v1/data/transferprocess" \
```

Comment:
this is not working, the `dataDestination` is missing the mandatory `type` field. Furthermore, no http server on port 4000 was launched.